### PR TITLE
Add flake8

### PR DESCRIPTION
**Pull request by andrewmil**
Add flake8 checks to repo

- Added a .flake file where we ignore line too long errors
- Added flake8 to the travis build (it comes built in with pipenv)
- Fixed flake8 failures
- Updated README with how to run flake8 through pipenv

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+ignore = E501
```
Comment (gemmai95) [inline .flake8]: Is there a reason the line length error is being ignored?

Comment (andrewmil) [reply .flake8]: I don't like it

Comment (andrewmil) [reply .flake8]: But seriously we should agree on these standards if we're going to change them, personally I think the line length limit is far too short and can discourage being explicit with function/variable names.

Comment (gemmai95) [reply .flake8]: You can configure it fairly easily: http://flake8.pycqa.org/en/2.6.0/config.html 
You can put it to what you want. The standard seems to be 160 max across SDC

Comment (andrewmil) [reply .flake8]: I'll try and implement that and see how many errors I get

Comment (gemmai95) [reply .flake8]: You can configure it to what is needed. I'm just not sure it should be ignored.

File: gunicorn.py [hunk 2]
```diff
@@ -200,15 +200,16 @@ def worker_int(worker):
...
+    import sys
+    import threading
+    import traceback
+
     id2name = dict([(th.ident, th.name) for th in threading.enumerate()])
     code = []
     for threadId, stack in sys._current_frames().items():
-        code.append("\n# Thread: %s(%d)" % (id2name.get(threadId,""),
-            threadId))
+        code.append("\n# Thread: %s(%d)" % (id2name.get(threadId, ""), threadId))
```
Comment (tej99) [inline gunicorn.py]: Is there a reason to not use .format() here? Feel like it would be cleaner and consistent with everywhere else.

Comment (andrewmil) [reply gunicorn.py]: I might be wrong but I think we copy/pasted the guincorn.py file from somewhere.
If we're wanting to refactor/tidy that file it's probably something to do in another PR. I just fixed the things that flake8 was complaining about here

**Pull request merged**